BDArmory/Control/MissileFire.cs

```diff
@@ -7213,19 +7196,11 @@ void BombAimer()
                 }
                 if (guardTarget)
                 {
-                    if ((prevPos - guardTarget.CoM).sqrMagnitude < closestPosSqr)
+                    float targetDist = Vector3.Distance(currPos, guardTarget.CoM) - guardTarget.GetRadius();
+                    if (targetDist < CurrentMissile.GetBlastRadius())
                     {
-                        var timeToCPA = AIUtils.ClosestTimeToCPA(guardTarget.CoM - prevPos, guardTarget.Velocity() - simVelocity, guardTarget.acceleration - gravity - dragForce, simDeltaTime);
-                        if (timeToCPA < simDeltaTime)
-                            closestPos = AIUtils.PredictPosition(prevPos, simVelocity, gravity + dragForce, timeToCPA);
-                        else
-                            closestPos = prevPos;
-                        closestPosSqr = (closestPos - guardTarget.CoM).sqrMagnitude;
-                    }
-                    else
-                    {
-                        bombAimerPosition = closestPos;
-                        break;
+                        bombAimerPosition = currPos;
+                        simulating = false;
```

Comment:
I think this simulation needs a more explicit exit condition in case none of the `simulating = false` conditions are met within a reasonable time.

Reply:
For lines 7197 - 7205 specifically, or for the whole simulation block in general?
What would be a reasonable time? There's a check in the code earlier to only do the sim if craft alt < 10km.

Reply:
The whole simulation block in general. I was getting stuck in an infinite loop from this before I merged in the latest commits, which took a while to figure out where it was coming from.
I think simply allowing 0.1s of real-time should be sufficient (that's what the ballistic trajectory sim uses), so something like
```csharp
var tic = Time.realtimeSinceStartup;
while (simulating && Time.realtimeSinceStartup - tic < 0.1f)
{...}
if (Time.realtimeSinceStartup - tic >= 0.1f)
{ /* simulation timed out */ }
```
or
```csharp
var tic = Time.realtimeSinceStartup;
while (simulating)
{
  ...
  if (Time.realtimeSinceStartup - tic > 0.1f)
  {
    /* simulation timed out */
    break;
  }
}
```